pkg/daemon/update.go

```diff
@@ -451,8 +451,10 @@ func (dn *Daemon) applyOSChanges(oldConfig, newConfig *mcfgv1.MachineConfig) (re
 	}()
 
 	// Apply kargs
-	if err := dn.updateKernelArguments(oldConfig, newConfig); err != nil {
-		return err
+	if mcDiff.kargs {
```

Comment:
Doesn't `dn.updateKerenlArguments` do a check already? 
https://github.com/openshift/machine-config-operator/blob/82929b72f6901650c5849ba4afe7ccdbed48b17c/pkg/daemon/update.go#L1018-L1021

Reply:
not really, generateKargsCommand(0 function just deletes all kargs available in old Rendered MC and appened all from new rendered MC. variable name is a bit misleading, perhaps it is hanging around from our earlier implementation. We have simplified kargs logic over time due to various bugs we hit, see https://github.com/openshift/machine-config-operator/pull/2014/files . I will update variable name so that we don't get confused

Reply:
Ah, okay.... let me know when you're ready for another review.

Reply:
updated